llm/llama-3/README.md

```diff
@@ -0,0 +1,340 @@
+IP=$(sky status --ip llama3)
+
+# We need to manually specify the stop_token_ids to make sure the model finish
+# on <|eot_id|>.
+curl http://$IP:8081/v1/chat/completions \
+  -H "Content-Type: application/json" \
+  -d '{
+    "model": "meta-llama/Meta-Llama-3-8B-Instruct",
+    "messages": [
+      {
+        "role": "system",
+        "content": "You are a helpful assistant."
+      },
+      {
+        "role": "user",
+        "content": "Who are you?"
+      }
+    ],
+    "stop_token_ids": [128009,  128001]
+  }'
+```
+
+To use the Gradio UI, open the URL shown in the logs:
+```console
+(task, pid=17433) Running on public URL: https://xxxxxxxxxx.gradio.live
+```
+
+Check the demo at the top of this page for an example of the Gradio UI.
+
+To stop the instance:
+```console
+sky stop llama3
+```
+
+To shut down all resources:
+```console
+sky down llama3
+```
+
+**Note**: If you would like to try the 8B model, you can use the following accelerators:
+```yaml
+resources:
+  accelerators: {L4, A10g, A10, L40, A40, A100, A100-80GB}
+```
+
+## Serving Llama-3: scaling up with SkyServe
+
+After playing with the model, you can deploy the model with autoscaling and load-balancing using SkyServe.
+
+With no change to the YAML, launch a fully managed service on your infra:
+```console
+HF_TOKEN=xxx sky serve up llama3.yaml -n llama3 --env HF_TOKEN
+```
+
+Wait until the service is ready:
+```console
+watch -n10 sky serve status llama3
+```
+
+<details>
+<summary>Example outputs:</summary>
+
+```console
+Services
+NAME  VERSION  UPTIME  STATUS  REPLICAS  ENDPOINT
+llama3  1        35s     READY   2/2       xx.yy.zz.100:30001
+
+Service Replicas
+SERVICE_NAME  ID  VERSION  IP            LAUNCHED     RESOURCES                       STATUS  REGION
+llama3          1   1        xx.yy.zz.121  18 mins ago  1x GCP([Spot]{'A100-80GB': 4})  READY   us-east4
+llama3          2   1        xx.yy.zz.245  18 mins ago  1x GCP([Spot]{'A100-80GB': 4})  READY   us-east4
+```
+</details>
+
+
+Get a single endpoint that load-balances across replicas:
+```console
+ENDPOINT=$(sky serve status --endpoint llama3)
+```
+
+> **Tip:** SkyServe fully manages the lifecycle of your replicas. For example, if a spot replica is preempted, the controller will automatically replace it. This significantly reduces the operational burden while saving costs.
+
+To curl the endpoint:
+```console
+curl -L $ENDPOINT/v1/chat/completions \
+  -H "Content-Type: application/json" \
+  -d '{
+    "model": "meta-llama/Meta-Llama-3-70B-Instruct",
+    "messages": [
+      {
+        "role": "system",
+        "content": "You are a helpful assistant."
+      },
+      {
+        "role": "user",
+        "content": "Who are you?"
+      }
+    ]
+  }'
+```
+
+To shut down all resources:
+```console
+sky serve down llama3
+```
+
+See more details in [SkyServe docs](https://skypilot.readthedocs.io/en/latest/serving/sky-serve.html).
+
+
+### **Optional**: Connect a GUI to your Llama-3 endpoint
```

Comment:
Maybe put the GUI gif here as well?

Reply:
It might be a bit duplicated and may have too many moving items in the page which may be a bit distracting. How about I put a skypilot+llama3 banner on the top and move that demo back here?

Reply:
SGTM.